.buildkite/pipelines/build_api_docs.yml

```diff
@@ -1,12 +1,32 @@
 env:
   PUBLISH_API_DOCS_CHANGES: 'true'
 steps:
+  - commands:
```

Comment:
We needed a major change in this pipeline due to the fact that on the new system, the issued github tokens are only valid for ~50 minutes, and this job would consistently get it expired before it could have committed the result.

The solution I found is to ran the typecheck and build in parallel, and make the build's commit&push wait for a status from the typecheck. It's demonstrated here: https://buildkite.com/elastic/kibana-migration-pipeline-staging/builds/70 ✅

Reply:
I think this is good and works, but obfuscates the real problem. I'm wondering if we need a better solution for this timeout because we might hit this in the future again?

Reply:
The other alternative would be to test+build, then archive the results, then collect the archive, and push in a new step. That might also work, but from what I understood from a discussion, the tokens are cached across steps, so it's not always guaranteed that we get a fresh token at the beginning of a step.

Here's the thread if you're interested in the details: https://elastic.slack.com/archives/C06N57G41HT/p1712766549810709

Reply:
Thanks for the context, definitely a bit tricky to work around. Maybe we could put the collect and push aspect into a new, generic pipeline which could be triggered by any other pipeline? Don't think it's necessary now unless we start hitting this issue often.

If you're happy to move forward with this, could we put it into a script so it follows the usual pattern? A comment on why you had to do this would be good as well.

Reply:
For now, I've only made the scripting change - for the future we can think if a generic push-oriented job is well grounded. I can see how it is a fitting component to our workflows, but I can't yet see the abuse potential and risks with it.

Reply:
I've noticed that even a 35m long step can have an outdated token, so I've implemented the upload in another step